packages/thirdweb/src/wallets/in-app/core/wallet/in-app-core.ts

```diff
@@ -228,5 +236,14 @@ export function createInAppWallet(args: {
       }
       emitter.emit("chainChanged", newChain);
     },
+    getAdminAccount: () => {
+      if (!account) {
+        // If we don't have an account at all, it's definitely not a smart account and it absolutely doesn't have an admin account
+        return undefined;
+      } else {
+        // This might return an admin account (if the main account is a smart account), otherwise it'll return undefined
+        return getAdminAccountForSmartAccount(account);
```

Comment:
How about we just store the admin account the same way we store the active account?

Would rather not rely on an in-memory map if possible.

That way you don't need the boolean flag either, just keep both accounts around

Reply:
Nice, I like that much better